mlir/unittests/Bytecode/BytecodeTest.cpp

```diff
@@ -148,3 +149,24 @@ TEST(Bytecode, OpWithoutProperties) {
   EXPECT_TRUE(OperationEquivalence::computeHash(op.get()) ==
               OperationEquivalence::computeHash(roundtripped));
 }
+
+TEST(Bytecode, FusedLocCrash) {
+  MLIRContext context;
+  OpBuilder builder(&context);
+  SmallVector<Location> locs;
+  FusedLoc fusedLoc = FusedLoc::get(&context, locs, {});
```

Comment:
Should we disallow this at the constructor? It feels weird to have a fused location with no locations. I don't know when one would actually want this behavior to be valid.

Reply:
Disallowing empty location looks reasonable but the issue itself is not specific to empty location. We should be able to get the same failure when fused loc has a single location that is not fused loc ([this](https://github.com/llvm/llvm-project/blob/af0f58cf14329f5e0fe56fed7c9eb4cd3107a1ce/mlir/lib/IR/Location.cpp#L119-L120) is the code) e.g:
```c++
FileLineColLoc loc = ...;
FusedLoc fusedLoc = FusedLoc::get(&context, {loc}, {}) 
```